db/migrate/20210504200438_add_github_updated_at_to_check_runs.rb

```diff
@@ -0,0 +1,5 @@
+class AddGithubUpdatedAtToCheckRuns < ActiveRecord::Migration[6.1]
+  def change
+    add_column :check_runs, :github_updated_at, :datetime, default: 0
```

Comment:
How precise are Github's timestamps? You can store datetime data down to the microsecond with some tweaking.

Reply:
GitHub's timestamps are to the second. From their example (other PR): https://github.com/Shopify/shipit-engine/blob/e8b5f82b5ea2689d3bf1c5b6b6d542e6bf9b15b6/test/fixtures/payloads/check_run_success.json#L13

Reply:
@casperisfine @ajshepley @gigr , #TIL Using `default: 0` to set a default calculated timestamp in MySQL only works in certain SQL modes - set by the server instance - and can cause errors when running migrations.

> The following examples use DEFAULT 0, a default that can produce warnings or errors depending on whether strict SQL mode or the NO_ZERO_DATE SQL mode is enabled. Be aware that the TRADITIONAL SQL mode includes strict mode and NO_ZERO_DATE.
-- https://dev.mysql.com/doc/refman/8.0/en/timestamp-initialization.html

Our MySQL server instances / clusters include the `NO_ZERO_DATE` SQL mode which means this migration fails without alteration. 

I _think_ the goal of `default: 0` is to set the github_updated_at value for new records to the current database time so that the check in `Shipit::CheckRun#create_or_update_by!`[^1] will avoid messaging nil, and indicate that Shipit _should_ save the CheckRun for the first webhook it receives. Am I close to the mark? I'm really interested in "Have I understood the intent / context correctly?"

In this case, for any folks like me who _might_ run into this issue, I _think_ the host application can  work around this failing migration by rewriting the default to use the `CURRENT_TIMESTAMP` instead of `0` like so:

```ruby
add_column :check_runs, :github_updated_at, :datetime , default: -> { "CURRENT_TIMESTAMP" }
```

References
----------

- https://dev.mysql.com/doc/refman/8.0/en/timestamp-initialization.html
- https://dev.mysql.com/doc/refman/8.0/en/sql-mode.html
- https://dev.mysql.com/doc/refman/5.7/en/timestamp-initialization.html

[^1]: https://github.com/Shopify/shipit-engine/blob/ff4300326a981b9036df3565b719352bbb75c114/app/models/shipit/check_run.rb#L25

Reply:
Either way, it look like this column is supposed to store GitHub data, so from a quick glance I feel like the default should have been `NULL`.